+++ boost/tools/build/src/tools/emscripten.jam
--- a/src/tools/emscripten.jam
@@ -52,7 +52,7 @@
         ;
 
 type.set-generated-target-suffix EXE : <toolset>emscripten : "js" ;
-type.set-generated-target-suffix OBJ : <toolset>emscripten : "bc" ;
+type.set-generated-target-suffix OBJ : <toolset>emscripten : "o" ;
 type.set-generated-target-suffix STATIC_LIB : <toolset>emscripten : "bc" ;
 
 toolset.flags emscripten.compile OPTIONS <flags> ;
