diff -ru ccp4/ccp4_sysdep.h /Users/stuart/devtools-09032022/co/libccp4/ccp4/ccp4_sysdep.h
+++ libccp4/ccp4/ccp4_sysdep.h	2022-03-09 14:27:20.000000000 +0000
--- /Users/stuart/devtools-09032022/co/libccp4/ccp4/ccp4_sysdep.h	2022-03-09 09:36:52.000000000 +0000
@@ -200,7 +200,7 @@
 #  define NATIVEIT DFNTI_IBO
 #endif
 
+#if defined(MIPSEL) || defined(i386) || defined(i860) || defined(__ia64__) || defined(__amd64__) || defined(__x86_64__) || defined(_M_AMD64) || defined(__EMSCRIPTEN__)
-#if defined(MIPSEL) || defined(i386) || defined(i860) || defined(__ia64__) || defined(__amd64__) || defined(__x86_64__) || defined(_M_AMD64)
 #  define NATIVEIT DFNTI_IBO
 #  define NATIVEFT DFNTF_LEIEEE
 #endif
