diff -ru coot-1.0/src/fix-nomenclature.cc /Users/stuart/devtools-09032022/coot-1.0/src/fix-nomenclature.cc
+++ coot-1.0/src/fix-nomenclature.cc	2022-03-22 18:18:36.000000000 +0000
--- /Users/stuart/coot/fix-nomenclature.cc	2022-03-22 18:19:00.000000000 +0000
@@ -40,7 +40,6 @@
       atom_selection_container_t asc = get_atom_selection(filename, true, false, false);
       coot::nomenclature n(asc.mol);
       asc.mol->WritePDBASCII(argv[2]);
-      n.fix(&geom);
    }
    return 0;
 }
