diff --git a/coot-utils/slurp-map.cc b/coot-utils/slurp-map.cc
index ba9efdecb..227f58997 100644
--- a/coot-utils/slurp-map.cc
+++ b/coot-utils/slurp-map.cc
@@ -23,6 +23,10 @@
 #include <chrono>
 #include <zlib.h>
 
+#ifdef  __EMSCRIPTEN__
+#include <emscripten/version.h>
+#endif
+
 #include "utils/coot-utils.hh"
 #include "slurp-map.hh"
 
@@ -112,10 +116,12 @@ coot::util::slurp_fill_xmap_from_map_file(const std::string &file_name,
       std::string ext = file_name_extension(file_name);
       if (ext == ".gz") is_gzip = true;
 
+#if !((__POINTER_WIDTH__==64) && __EMSCRIPTEN__) || (__EMSCRIPTEN_major__ >=3 && __EMSCRIPTEN_minor__ >=1 && __EMSCRIPTEN_tiny__ >=63)
       if (is_gzip) {
          // this can fail (at the moment) if the axes are not in X,Y,Z order
          status = slurp_fill_xmap_from_gz_map_file(file_name, xmap_p, check_only);
       } else {
+#endif
          struct stat s;
          int fstat = stat(file_name.c_str(), &s);
          if (fstat == 0) {
@@ -145,7 +151,9 @@ coot::util::slurp_fill_xmap_from_map_file(const std::string &file_name,
                std::cout << "WARNING:: out-of-memory " << st_size+1 << " " << e.what() << std::endl;
             }
          }
+#if !((__POINTER_WIDTH__==64) && __EMSCRIPTEN__) || (__EMSCRIPTEN_major__ >=3 && __EMSCRIPTEN_minor__ >=1 && __EMSCRIPTEN_tiny__ >=63)
       }
+#endif
    } else {
       std::cout << "WARNING:: file does not exist " << file_name << std::endl;
    }
