--- coot-1.0/compat/getopt.cc	2022-03-23 15:13:07.000000000 +0000
+++ coot-1.0/compat/getopt.cc-patched	2022-03-23 15:12:59.000000000 +0000
@@ -211,6 +212,12 @@
    `first_nonopt' and `last_nonopt' are relocated so that they describe
    the new indices of the non-options in ARGV after they are moved.  */

+void clear_getopt_initialized(){
+    __getopt_initialized = 0;
+    coot_optind = 1;
+    first_nonopt = 0;
+    last_nonopt = 0;
+}
