diff --git a/coot-utils/slurp-map.cc b/coot-utils/slurp-map.cc
index b57e1c037..f1dd0cd2c 100644
--- a/coot-utils/slurp-map.cc
+++ b/coot-utils/slurp-map.cc
@@ -112,10 +112,12 @@ coot::util::slurp_fill_xmap_from_map_file(const std::string &file_name,
       std::string ext = file_name_extension(file_name);
       if (ext == ".gz") is_gzip = true;
 
+#if !((__POINTER_WIDTH__==64) && __EMSCRIPTEN__)
       if (is_gzip) {
          // this can fail (at the moment) if the axes are not in X,Y,Z order
          status = slurp_fill_xmap_from_gz_map_file(file_name, xmap_p, check_only);
       } else {
+#endif
          struct stat s;
          int fstat = stat(file_name.c_str(), &s);
          if (fstat == 0) {
@@ -145,7 +147,9 @@ coot::util::slurp_fill_xmap_from_map_file(const std::string &file_name,
                std::cout << "WARNING:: out-of-memory " << st_size+1 << " " << e.what() << std::endl;
             }
          }
+#if !((__POINTER_WIDTH__==64) && __EMSCRIPTEN__)
       }
+#endif
    } else {
       std::cout << "WARNING:: file does not exist " << file_name << std::endl;
    }
