=== modified file 'src/gesamt.cpp'
--- old/src/gesamt.cpp	2017-03-01 15:22:09 +0000
+++ gesamt/src/gesamt.cpp	2022-05-04 14:32:21 +0000
@@ -27,6 +27,10 @@
 // =================================================================
 //
 
+#if defined(__EMSCRIPTEN__)
+#include <vector>
+#include <string>
+#endif
 #include <string.h>
 
 #include "gesamt_input.h"
@@ -165,8 +169,23 @@
 
 }
 
+#if defined(_CCP4_WEB_EXAMPLE_)
+int gesamt_main(const std::vector<std::string> &_argv)  {
+#else
 int main ( int argc, char ** argv, char ** env )  {
+#if defined(__EMSCRIPTEN__)
+#ifndef NODERAWFS
+    // mount the current folder as a NODEFS instance
+    // inside of emscripten
+    EM_ASM(
+            FS.mkdir('/working');
+            FS.mount(NODEFS, { root: '.' }, '/working');
+          );
+#endif
+#else
 UNUSED_ARGUMENT(env);
+#endif
+#endif
 gsmt::Input       Input;
 #ifdef compile_for_ccp4
 mmdb::pstr        ccp4msg = NULL;
@@ -195,8 +214,18 @@
 
 #endif
 
+#if defined(_CCP4_WEB_EXAMPLE_)
+    const char *progName =  "web_superpose";
+#endif
+#if defined(_CCP4_WEB_EXAMPLE_)
+  int argc = _argv.size() + 1;
+#endif
   if (argc<=1)  {
+#if defined(_CCP4_WEB_EXAMPLE_)
+    printInstructions ( (char*)progName );
+#else
     printInstructions ( argv[0] );
+#endif
 #ifdef compile_for_ccp4
     printf ( "<!--SUMMARY_END-->\n" );
     ccperror ( 1,"No input" );
@@ -204,6 +233,7 @@
     return 1;
   }
 
+#if !defined(_CCP4_WEB_EXAMPLE_)
   if ((!strcmp(argv[1],"-?")) || (!strcasecmp(argv[1],"-help")) ||
       (!strcasecmp(argv[1],"--help")))  {
     printInstructions ( argv[0] );
@@ -213,15 +243,22 @@
 #endif
     return 2;
   }
+#endif
 
+#if !defined(_CCP4_WEB_EXAMPLE_)
   if (!strcmp(argv[1],"--print-opt"))  {
     printViewerOptTemplate ( argv[0] );
     return 0;
   }
+#endif
 
 
   if (argc<3)  {
+#if defined(_CCP4_WEB_EXAMPLE_)
+    printInstructions ( (char*)progName );
+#else
     printInstructions ( argv[0] );
+#endif
 #ifdef compile_for_ccp4
     printf ( "<!--SUMMARY_END-->\n" );
     ccperror ( 3,"Wrong input" );
@@ -229,7 +266,11 @@
     return 3;
   }
   
-  IC = Input.parseCommandLine ( argc,argv );
+#if defined(_CCP4_WEB_EXAMPLE_)
+  IC = Input.parseCommandLine ( _argv);
+#else
+  IC = Input.parseCommandLine ( argc, argv );
+#endif
   if (IC==gsmt::INPUT_Ok)  {
 
     switch (Input.taskCode)  {
@@ -265,7 +306,11 @@
     printf ( "\n\n"
              " *** INPUT ERROR (%i): %s\n\n"
              "     Run %s -help for full list of options.\n\n",
+#if defined(_CCP4_WEB_EXAMPLE_)
+             (int)IC,Input.errMessage(IC),(char*)progName );
+#else
              (int)IC,Input.errMessage(IC),argv[0] );
+#endif
 #ifdef compile_for_ccp4
     printf ( "<!--SUMMARY_END-->\n" );
     mmdb::CreateCopy ( ccp4msg,Input.errMessage(IC) );

=== modified file 'src/gesamt_input.cpp'
--- old/src/gesamt_input.cpp	2021-05-06 13:16:40 +0000
+++ gesamt/src/gesamt_input.cpp	2022-05-04 14:27:14 +0000
@@ -181,11 +181,23 @@
 }
 
 
+#if defined(_CCP4_WEB_EXAMPLE_)
+gsmt::INPUT_CODE gsmt::Input::parseCommandLine (
+                                           const std::vector<std::string> &_argv )  {
+#else
 gsmt::INPUT_CODE gsmt::Input::parseCommandLine (
                                            int argc, char ** argv )  {
+#endif
 int   argNo,argc1;
 bool  done;
   
+#if defined(_CCP4_WEB_EXAMPLE_)
+  int argc = _argv.size();
+  char **argv = new char*[argc];
+  for(int i=0;i<argc;i++){
+      argv[i] = (char *)(_argv[i].c_str());
+  }
+#endif
   freeMemory();
   
   mmdb::CreateCopy ( rvapiJSUri,"jsrview" );

=== modified file 'src/gesamt_input.h'
--- old/src/gesamt_input.h	2017-03-01 15:22:09 +0000
+++ gesamt/src/gesamt_input.h	2022-05-04 14:21:16 +0000
@@ -27,6 +27,11 @@
 #ifndef __GSMT_Input__
 #define __GSMT_Input__
 
+#if defined(__EMSCRIPTEN__)
+#include <vector>
+#include <string>
+#endif
+
 #include "mmdb2/mmdb_mattype.h"
 #include "gesamtlib/gsmt_viewer.h"
 #include "gesamtlib/gsmt_sheaf.h"
@@ -122,7 +127,11 @@
       Input ();
       ~Input();
 
+#if defined(_CCP4_WEB_EXAMPLE_)
+      INPUT_CODE parseCommandLine ( const std::vector<std::string> &argv );
+#else
       INPUT_CODE parseCommandLine ( int argc, char ** argv );
+#endif
       
       mmdb::cpstr errMessage ( INPUT_CODE icode );
       

=== modified file 'src/gesamt_pairwise.cpp'
--- old/src/gesamt_pairwise.cpp	2020-01-14 16:01:51 +0000
+++ gesamt/src/gesamt_pairwise.cpp	2022-05-04 14:33:46 +0000
@@ -147,7 +147,7 @@
 }
 
 
-void printFracAnalysis ( mmdb::mat44 & T, mmdb::cpstr name,
+void printFracAnalysisGSMT ( mmdb::mat44 & T, mmdb::cpstr name,
                          mmdb::PManager M )   {
 mmdb::mat44 TF;
 

=== modified file 'src/gesamt_pairwise.h'
--- old/src/gesamt_pairwise.h	2016-06-14 17:03:26 +0000
+++ gesamt/src/gesamt_pairwise.h	2022-05-04 14:34:00 +0000
@@ -37,7 +37,7 @@
 
 extern void alignToTarget         ( gsmt::RInput Input );
 
-extern void printFracAnalysis     ( mmdb::mat44 & T, mmdb::cpstr name,
+extern void printFracAnalysisGSMT     ( mmdb::mat44 & T, mmdb::cpstr name,
                                     mmdb::PManager M );
 extern void writeFracAnalysis_csv ( mmdb::pstr S,
                                     mmdb::mat44 & T, mmdb::cpstr name,

