--- privateer-wasm/src/privateer/cpp/privateer-json.h	2024-12-17 10:42:56
+++ privateer-wasm/src/privateer/cpp/privateer-json.h	2024-12-17 10:43:21
@@ -11,8 +11,7 @@
 
 #include <iostream>
 #include "privateer-error.h"
-// #include "third-party/sajson.h" // Have to include it straight from gemmi, as otherwise it leads to "error: multiple definition of ‘enum sajson::type’" errors.
-#include "gemmi/third_party/sajson.h" 
+#include "third-party/sajson.h"
 #include <vector>
 #include <unordered_map>
 #include <cstdio>    // for FILE, fopen, fclose

