=== modified file 'superpose.cpp'
--- old/superpose.cpp	2017-05-26 23:01:01 +0000
+++ ssm/superpose.cpp	2022-05-04 14:29:58 +0000
@@ -51,6 +51,10 @@
 // =================================================================
 //
 
+#if defined(__EMSCRIPTEN__)
+#include <vector>
+#include <string>
+#endif
 #include <string.h>
 
 #include "ssm/ssm_align.h"
@@ -68,6 +72,17 @@
 
 #endif
 
+#if defined(__EMSCRIPTEN__)
+#include <emscripten.h>
+
+#ifdef NODERAWFS
+#define CWD ""
+#else
+#define CWD "/working/"
+#endif
+#else
+#define CWD ""
+#endif
 
 #ifdef _emulate_ccp4_
 
@@ -105,7 +120,7 @@
 #endif
 
 
-void printInstructions ( char *argv0 )  {
+void printInstructionsSSM ( char *argv0 )  {
 
   printf (
     "\n"
@@ -255,8 +270,23 @@
 }
 */
 
+#if defined(_CCP4_WEB_EXAMPLE_)
+int superpose_main(const std::vector<std::string> &files, const std::vector<std::string> &selections)  {
+#else
 int main ( int argc, char ** argv, char ** env )  {
+#if defined(__EMSCRIPTEN__)
+#ifndef NODERAWFS
+    // mount the current folder as a NODEFS instance
+    // inside of emscripten
+    EM_ASM(
+            FS.mkdir('/working');
+            FS.mount(NODEFS, { root: '.' }, '/working');
+          );
+#endif
+#else
 UNUSED_ARGUMENT(env);
+#endif
+#endif
 mmdb::PPManager M;
 mmdb::psvector  name;
 mmdb::psvector  selstring;
@@ -269,11 +299,19 @@
   ccp4_banner();
 #endif
 
+#if defined(_CCP4_WEB_EXAMPLE_)
+  int argc = files.size() + 1;
+#endif
   if (argc<=1)  {
 #ifdef _ccp4_
     printf ( "<!--SUMMARY_BEGIN-->\n" );
 #endif
-    printInstructions ( argv[0] );
+#if defined(_CCP4_WEB_EXAMPLE_)
+    const char *progName =  "web_superpose";
+    printInstructionsSSM ( (char*)progName );
+#else
+    printInstructionsSSM ( argv[0] );
+#endif
 #ifdef _ccp4_
     printf ( "<!--SUMMARY_END-->\n" );
     ccperror ( 1,"No input" );
@@ -281,18 +319,20 @@
     return 1;
   }
 
+#if !defined(_CCP4_WEB_EXAMPLE_)
 #ifdef _ccp4_
   printf ( "<!--SUMMARY_BEGIN-->\n" );
 #endif
   if ((!strcmp(argv[1],"-?")) || (!strcasecmp(argv[1],"-help")) ||
       (!strcasecmp(argv[1],"--help")))  {
-    printInstructions ( argv[0] );
+    printInstructionsSSM ( argv[0] );
 #ifdef _ccp4_
     printf ( "<!--SUMMARY_END-->\n" );
     ccperror ( 2,"Wrong input" );
 #endif
     return 2;
   }
+#endif
 
   printf ( "\n"
     " Superpose v." superpose_version " from " superpose_date " "
@@ -323,22 +363,39 @@
   argNo = 1;
   rc    = 0;
   while ((argNo<argc) && (!rc))  {
+#if !defined(_CCP4_WEB_EXAMPLE_)
     if (!strcasecmp(argv[argNo],"-o"))  {
       argNo++;
       if (argNo<argc)
         mmdb::CreateCopy ( fileout,argv[argNo++] );
     } else  {
+#endif
+#if defined(_CCP4_WEB_EXAMPLE_)
+      char *fn = (char*)files[argNo-1].c_str();
+      mmdb::CreateCopy ( name[nStructures],fn );
+#else
       mmdb::CreateCopy ( name[nStructures],argv[argNo] );
-      if (readCoorFile(argv[argNo++],M[nStructures]))
+      char *fn =(char*) malloc(strlen(CWD)+strlen(argv[argNo]));
+      fn[0] = '\0';
+      strncat(fn,CWD,strlen(CWD));
+      strncat(fn,argv[argNo],strlen(CWD));
+#endif
+      argNo++;
+
+      if (readCoorFile(fn ,M[nStructures]))
         rc = 3;
+#if !defined(_CCP4_WEB_EXAMPLE_)
       else if (argNo<argc)  {
         if (selectAtoms(M[nStructures],argv,argNo,
                 selstring[nStructures],selHnd[nStructures]))
           rc = 4;
       } else
         mmdb::CreateCopy ( selstring[nStructures],"*" );
+#endif
       nStructures++;
+#if !defined(_CCP4_WEB_EXAMPLE_)
     }
+#endif
   }
 
   if (!rc)  {

